### DISCUSSION TITLE
[FEATURE REQUEST] Make use of ID_LIKE from /etc/os-release for wider OS derivatives coverage

### DISCUSSION THREAD

**Discussion by ScriptAutomate** [saltstack/salt#67482] [Feature Requests]
**Is your feature request related to a problem? Please describe.**

A PR was opened to include coverage for Pop!_OS 20.04: #58443

By default, we have to hardcode many OS ID's into Salt in order for them to be covered when it comes to things like package management

**Describe the solution you'd like**

Though, Pop!_OS 20.04 has the following in `/etc/os-release`:

```
NAME="Pop!_OS"
VERSION="20.04 LTS"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 20.04 LTS"
VERSION_ID="20.04"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
LOGO=distributor-logo-pop-os
```

Where the `ID_LIKE` option is of importance (as pointed out by @ids1024).

Linux man pages definition of `ID_LIKE` [as seen in the OPTIONS section](https://www.man7.org/linux/man-pages/man5/os-release.5.html#OPTIONS):

```
       ID_LIKE=
           A space-separated list of operating system identifiers in the
           same syntax as the ID= setting. It should list identifiers of
           operating systems that are closely related to the local operating
           system in regards to packaging and programming interfaces, for
           example listing one or more OS identifiers the local OS is a
           derivative from. An OS should generally only list other OS
           identifiers it itself is a derivative of, and not any OSes that
           are derived from it, though symmetric relationships are possible.
           Build scripts and similar should check this variable if they need
           to identify the local operating system and the value of ID= is
           not recognized. Operating systems should be listed in order of
           how closely the local operating system relates to the listed
           ones, starting with the closest. This field is optional. Example:
           for an operating system with "ID=centos", an assignment of
           "ID_LIKE="rhel fedora"" would be appropriate. For an operating
           system with "ID=ubuntu", an assignment of "ID_LIKE=debian" is
           appropriate.
```

We should evaluate having a conditional check for the `ID_LIKE` list of values when an OS ID isn't currently hardcoded as supported by Salt via `_OS_NAME_MAP` and `_OS_FAMILY_MAP`. In this case, Pop!_OS would have been able to be managed without the need of hardcoding the values, as a lookup could have been done based on the `ID_LIKE` value in `/etc/os-release`

**Comment by s0undt3ch** [comment]
We now depend on distro, and distro has a `.like()` method(Or similar) which we could use instead of doing our own parsing.

**Comment by s0undt3ch** [comment]
https://distro.readthedocs.io/en/latest/#distro.like

**Comment by bdrung** [comment]
`distro.like()` provides the data from `ID_LIKE` from the os-release file. So since we already parse os-release, we have the data and can use it. Side note: Python 3.10 added support for reading os-release and salt should use it (if available): https://github.com/saltstack/salt/pull/61589

**Comment by bdrung** [comment]
I worked on cleaning up the OS grain calculation on Linux in https://github.com/saltstack/salt/pull/61626. This merge request contains the first step in this direction. The new code derives the `os_family` grain from the `ID_LIKE` field from os-release for selected distributions (to not cause regressions). To implement this feature, the new code needs to be used by default (and not just for selected distributions).
